programs/fileio.c

```diff
@@ -862,6 +863,205 @@ int FIO_compressFilename(const char* dstFileName, const char* srcFileName,
+                    int const ret = fseek(srcFile, (long)frameSize, SEEK_CUR);
+                    if (ret != 0) {
+                        DISPLAY("Error: could not find end of skippable frame\n");
+                        detectError = 1;
+                        break;
+                    }
+                }
+                info->numSkippableFrames++;
+            }
+            else {
+                detectError = 2;
+                break;
+            }
+        }
+    }
+    fclose(srcFile);
+    return detectError;
+}
+
+static void displayInfo(const char* inFileName, fileInfo_t* info, int displayLevel){
+    double const compressedSizeMB = (double)info->compressedSize/(1 MB);
+    double const decompressedSizeMB = (double)info->decompressedSize/(1 MB);
+    const char* checkString = (info->usesCheck ? "XXH64" : "None");
+    if (displayLevel <= 2) {
+        if (info->canComputeDecompSize) {
+            double const ratio = (info->decompressedSize == 0) ? 0.0 : compressedSizeMB/decompressedSizeMB;
+            DISPLAYOUT("Skippable  Non-Skippable  Compressed  Uncompressed  Ratio  Check  Filename\n");
+            DISPLAYOUT("%9d  %13d  %7.2f MB  %9.2f MB  %5.3f  %s  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, decompressedSizeMB,
+                    ratio, checkString, inFileName);
+        }
+        else {
+            DISPLAYOUT("Skippable  Non-Skippable  Compressed  Check  Filename\n");
+            DISPLAYOUT("%9d  %13d  %7.2f MB  %s  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, checkString, inFileName);
+        }
+    }
+    else{
+        DISPLAYOUT("# Zstandard Frames: %d\n", info->numActualFrames);
+        DISPLAYOUT("# Skippable Frames: %d\n", info->numSkippableFrames);
+        DISPLAYOUT("Compressed Size: %.2f MB (%llu B)\n", compressedSizeMB, info->compressedSize);
+        if (info->canComputeDecompSize) {
+            DISPLAYOUT("Decompressed Size: %.2f MB (%llu B)\n", decompressedSizeMB, info->decompressedSize);
+            DISPLAYOUT("Ratio: %.4f\n", compressedSizeMB/decompressedSizeMB);
+        }
+        if (info->usesCheck) {
+            DISPLAYOUT("Check: XXH64\n");
+        }
+    }
+
+    DISPLAYOUT("\n");
+}
+
+int FIO_listFile(const char* inFileName, int displayLevel){
+    fileInfo_t info;
+
+    /* initialize info to avoid warnings */
+    info.numActualFrames = 0;
+    info.numSkippableFrames = 0;
+    info.decompressedSize = 0;
+    info.canComputeDecompSize = 1;
+    info.compressedSize = 0;
+    info.usesCheck = 0;
+    DISPLAYOUT("File: %s\n", inFileName);
+    {
+        int const error = getFileInfo(&info, inFileName);
+        if (error == 1) {
+            /* display error, but provide output */
+            DISPLAY("An error occurred with getting file info\n");
+        }
+        else if (error == 2) {
+            DISPLAYOUT("File %s not compressed with zstd\n\n", inFileName);
```

Comment:
At the moment, --list prints a space between consecutive files when outputting information, so the double `\n\n` is so that when a file is not compressed with zstd it still prints a line between that file and the next. Should I just get rid of the line in between files?